packages/server-wallet/src/db/migrations/20201104115400_lower_case_constraints.ts

```diff
@@ -0,0 +1,52 @@
+import * as Knex from 'knex';
+
+import {addAddressCheck, addBytes32Check, addBytesCheck, dropConstraint} from '../utils';
+
+import {
+  dropByteConstraints as dropOldByteConstraints,
+  addByteContraints as addOldByteContraints,
+} from './20200707165856_initial';
+
+const channels = 'channels';
+const signingWallets = 'signing_wallets';
+const app_bytecode = 'app_bytecode';
+
+export async function addByteContraints(knex: Knex): Promise<void> {
+  await addBytes32Check(knex, signingWallets, 'private_key');
+  await addAddressCheck(knex, signingWallets, 'address');
+
+  await addBytes32Check(knex, channels, 'channel_id');
+  await addAddressCheck(knex, channels, 'app_definition');
+  await addAddressCheck(knex, channels, 'signing_address');
+  await addBytes32Check(knex, channels, 'funding_ledger_channel_id');
+
+  await addAddressCheck(knex, app_bytecode, 'app_definition');
+  await addBytesCheck(knex, app_bytecode, app_bytecode);
+
+  await addAddressCheck(knex, 'funding', 'asset_holder');
+}
+
+export async function dropByteConstraints(knex: Knex): Promise<void> {
+  await dropConstraint(knex, signingWallets, 'private_key_is_bytes32');
+  await dropConstraint(knex, signingWallets, 'address_is_address');
+
+  await dropConstraint(knex, channels, 'channel_id_is_bytes32');
+  await dropConstraint(knex, channels, 'app_definition_is_address');
+  await dropConstraint(knex, channels, 'signing_address_is_address');
+  await dropConstraint(knex, channels, 'funding_ledger_channel_id_is_bytes32');
+
+  await dropConstraint(knex, app_bytecode, 'app_definition_is_address');
+  await dropConstraint(knex, app_bytecode, 'app_bytecode_is_bytes');
+
+  await dropConstraint(knex, 'funding', 'asset_holder_is_address');
+}
+
+export async function up(knex: Knex): Promise<any> {
+  await dropOldByteConstraints(knex);
+  await addByteContraints(knex);
```

Comment:
Should this include an operation to lowercase any existing columns? Otherwise it looks like the migration could fail here.

Reply:
That's a good point. In fact, the migration will fail if there is existing data as, before this PR, addresses were checksummed.

Reply:
Now that addresses are checksummed, I don't think this comment is applicable.